src/fsipy/automatedPostprocessing/postprocessing_h5py/postprocessing_common_h5py.py

```diff
@@ -0,0 +1,470 @@
+        output_folder (Union[str, Path]): Path to the output folder where the transformed matrix will be stored.
+        mesh_path (Union[str, Path]): Path to the input mesh data.
+        case_name (str): Name of the simulation case.
+        start_t (float): Start time for extracting data.
+        end_t (float): End time for extracting data.
+        dvp (str): Quantity to extract (e.g., 'd' for displacement, 'v' for velocity).
+        stride (int): Stride for selecting timesteps.
+
+    Returns:
+        float: Time between simulation output files.
+    """
+    logging.info(f"Creating matrix for case {case_name}...")
+    input_path = Path(input_path)
+    output_folder = Path(output_folder)
+
+    # Create output directory if it doesn't exist
+    if not output_folder.exists():
+        output_folder.mkdir(parents=True)
+        logging.info(f'Output directory created: {output_folder}')
+    else:
+        logging.info(f'Output directory already exists: {output_folder}')
+
+    # Get node ID's from input mesh. If save_deg=2, you can supply the original mesh to get the data for the
+    # corner nodes, or supply a refined mesh to get the data for all nodes (very computationally intensive)
+    if dvp in {"d", "v", "p"}:
+        fluid_ids, wall_ids, all_ids = get_domain_ids(mesh_path)
+        ids = all_ids
+
+    # Get name of xdmf file
+    xdmf_files = {
+        'd': 'displacement.xdmf',
+        'v': 'velocity.xdmf',
+        'p': 'pressure.xdmf',
+        'wss': 'WSS_ts.xdmf',
+        'mps': 'MaxPrincipalStrain.xdmf',
+        'strain': 'InfinitesimalStrain.xdmf'
+    }
+
+    if dvp in xdmf_files:
+        xdmf_path = input_path / xdmf_files[dvp]
+    else:
+        raise ValueError("Invalid value for dvp. Please use 'd', 'v', 'p', 'wss', 'mps', or 'strain'.")
+
+    # If the simulation has been restarted, the output is stored in multiple files and may not have even
+    # temporal spacing.
+    # This loop determines the file names from the xdmf output file
+    with xdmf_path.open('r') as file1:
+        lines = file1.readlines()
+
+    h5_ts = []
+    time_ts = []
+    index_ts = []
+
+    # This loop goes through the xdmf output file and gets the time value (time_ts),
+    # associated .h5 file (h5_ts), and index of each timestep in the corresponding h5 file (index_ts)
+    for line in lines:
+        if '<Time Value' in line:
+            time_pattern = '<Time Value="(.+?)"'
+            time_str = re.findall(time_pattern, line)
+            time = float(time_str[0])
+            time_ts.append(time)
+
+        elif 'VisualisationVector' in line:
+            h5_pattern = '"HDF">(.+?):/'
+            h5_str = re.findall(h5_pattern, line)
+            h5_ts.append(h5_str[0])
+
+            index_pattern = "VisualisationVector/(.+?)</DataItem>"
+            index_str = re.findall(index_pattern, line)
+            index = int(index_str[0])
+            index_ts.append(index)
+
```

Comment:
This is also implemented as a function here 
https://github.com/KVSlab/Aneurysm_Workflow_FSI/blob/d98e4c73b9136592df3b06214a1b31ee2ee31b58/src/fsipy/automatedPostprocessing/postprocessing_fenics/create_hdf5.py#L213